packages/protocol/test/mainnet_forking/blockchain_scripts/migrations/v2.4/migrate.test.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  DISTRIBUTOR_ROLE,
+  getProtocolOwner,
+  getTruffleContract,
+  ZERO_ADDRESS,
+} from "packages/protocol/blockchain_scripts/deployHelpers"
+import {fundWithWhales} from "@goldfinch-eng/protocol/blockchain_scripts/helpers/fundWithWhales"
+
+import * as migrate2_4 from "@goldfinch-eng/protocol/blockchain_scripts/migrations/v2.4/migrate"
+import {expectOwnerRole, expectProxyOwner, expectRoles} from "@goldfinch-eng/protocol/test/testHelpers"
+import {TEST_TIMEOUT} from "../../../MainnetForking.test"
+import {impersonateAccount} from "@goldfinch-eng/protocol/blockchain_scripts/helpers/impersonateAccount"
+import {isMerkleDistributorInfo} from "@goldfinch-eng/protocol/blockchain_scripts/merkle/merkleDistributor/types"
+import {isMerkleDirectDistributorInfo} from "@goldfinch-eng/protocol/blockchain_scripts/merkle/merkleDirectDistributor/types"
+import BN from "bn.js"
+import {
+  BackerMerkleDirectDistributorInstance,
+  CommunityRewardsInstance,
+  GFIInstance,
+} from "@goldfinch-eng/protocol/typechain/truffle"
+
+const setupTest = deployments.createFixture(async () => {
+  await deployments.fixture("base_deploy", {keepExistingDeployments: true})
+
+  const gfi = await getTruffleContract<GFIInstance>("GFI")
+  const communityRewards = await getTruffleContract<CommunityRewardsInstance>("CommunityRewards")
+  const communityRewardsStartingBalance = await communityRewards.rewardsAvailable()
+
+  const {gf_deployer} = await getNamedAccounts()
+  assertIsString(gf_deployer)
+  await fundWithWhales(["ETH"], [gf_deployer])
+  await impersonateAccount(hre, await getProtocolOwner())
+  await fundWithWhales(["ETH"], [await getProtocolOwner()])
+
+  await migrate2_4.main()
+
+  return {
+    gfi,
+    communityRewards,
+    communityRewardsStartingBalance,
+  }
+})
+
+describe("v2.4", async function () {
+  this.timeout(TEST_TIMEOUT)
+
+  let gfi: GFIInstance
+  let communityRewards: CommunityRewardsInstance
+  let communityRewardsStartingBalance: BN
+
+  beforeEach(async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({gfi, communityRewards, communityRewardsStartingBalance} = await setupTest())
+    console.log("communityRewardsStartingBalance", communityRewardsStartingBalance.toString())
+  })
+
+  it("deploys BackerMerkleDirectDistributor", async () => {
+    await expect(deployments.get("BackerMerkleDirectDistributor")).to.not.be.rejected
+  })
+
+  it("initializes BackerMerkleDirectDistributor", async () => {
+    await expect(
+      (
+        await getTruffleContract<BackerMerkleDirectDistributorInstance>("BackerMerkleDirectDistributor")
+      ).initialize(ZERO_ADDRESS, ZERO_ADDRESS, web3.utils.keccak256("test"), {from: await getProtocolOwner()})
+    ).to.be.rejectedWith(/initialized/)
+  })
+
+  it("deploys BackerMerkleDistributor", async () => {
+    await expect(deployments.get("BackerMerkleDistributor")).to.not.be.rejected
+  })
+
+  expectProxyOwner({
+    toBe: async () => getProtocolOwner(),
```

Comment:
nit: you can just do `getProtocolOwner` here (and below) without the anon function